src/string_scanner.cr

```diff
@@ -280,6 +282,34 @@ class StringScanner
     @str.byte_slice(@byte_offset, @str.bytesize - @byte_offset)
   end
 
+  # Returns one byte from the current offset.
+  # ```
+  # require "string_scanner"
+  #
+  # s = StringScanner.new("あ")
+  # s.read_byte # => "\xE3"
+  # s.read_byte # => "\x81"
+  # s.read_byte # => "\x82"
+  # ```
+  def read_byte : String?
+    return nil if eos?
+    s = @str.byte_slice(@byte_offset, 1)
+    @byte_offset += 1
+    s
+  end
+
+  # Returns one char from the current offset.
+  # ```
+  # require "string_scanner"
+  #
+  # s = StringScanner.new("ab")
+  # s.read_char # => "a"
+  # s.read_char # => "b"
+  # ```
+  def read_char : String?
```

Comment:
I find it a bit unintuitive that `read_char` returns a `String?` and not a `Char?`.

Reply:
changed to return `Char?`.